### PR TITLE
IGNITE-6880: KNN Regression and Classification

### PR DESCRIPTION
**Pull request by zaleslaw**
Also added Labeled Dataset and loading from txt file

### CODE REVIEW

**Comment by oignatenko** [comment]
Please consider generifying LabeledDataset by using same type parameters as of underlying LabeledVector. Per my reading of code in LabeledDatasetTest this could lead to more type safe code. Such a change would probably better be done per separate ticket, because per my trial attempt to do this it may involve quite a lot of modifications - in particular member "data" would better change from array to list

File: modules/ml/src/main/java/org/apache/ignite/ml/knn/models/KNNModelFormat.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ */
+
+package org.apache.ignite.ml.knn.models;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import org.apache.ignite.ml.math.distances.DistanceMeasure;
+import org.apache.ignite.ml.structures.LabeledDataset;
+
+/** */
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/knn/models/KNNModelFormat.java]: Please add javadoc

Comment (ybabak) [reply modules/ml/src/main/java/org/apache/ignite/ml/knn/models/KNNModelFormat.java]: And same for getters

File: modules/ml/src/main/java/org/apache/ignite/ml/knn/models/KNNModel.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
+/*
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/knn/models/KNNModel.java]: no package-info for package org.apache.ignite.ml.knn.models. Please add it

File: modules/ml/src/main/java/org/apache/ignite/ml/math/distributed/keys/impl/SparseMatrixKey.java [hunk 3]
```diff
@@ -17,17 +17,18 @@
 
 package org.apache.ignite.ml.math.distributed.keys.impl;
 
-import java.io.Externalizable;
-import java.io.IOException;
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/math/distributed/keys/impl/SparseMatrixKey.java]: please check import format

File: modules/ml/src/main/java/org/apache/ignite/ml/structures/LabeledDataset.java [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+
+    /**
+     * Creates new Labeled Dataset by matrix and vector of labels
+     * @param mtx Given matrix with rows as observations
+     * @param lbs Labels of observations
+     * @param featureNames Column names
+     * @param isDistributed Use distributed data structures to keep data
+     */
+    public LabeledDataset(double[][] mtx, double[] lbs, String[] featureNames, boolean isDistributed) {
+        assert mtx != null;
```
Comment (ybabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/structures/LabeledDataset.java]: please format method

**Pull request closed without merging**